### PR TITLE
Add amino proto annotations

### PR DESCRIPTION
**Pull request by mattverse**
Closes: https://github.com/osmosis-labs/osmosis/pull/3782

Thanks @pyramation for pushing this forward!

## What is the purpose of the change
Adds amino proto annotations.
Detailed explanation on the benefits of this PR is mentioned in the original PR posted by @pyramation

TLDR by adding amino annotations, this would make many of the client side developers happy.

Note that this change required go mod changes, specifically sdk dependency. 
Previously the sdk dependency was somewhere in the commit of v15.x, whilst now I have branched off of v15.x, created v16.x including this change and token factory changes. 

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by mattverse** [comment]
This is blocked on https://github.com/osmosis-labs/osmosis/pull/4590 due to dependency issues, converting to draft for now

**Comment by mattverse** [comment]
Huh, why are the CIs frozen? 🤔

File: go.mod
Comment (czarcas7ic) [inline go.mod:318]: I cant find this commit on the v15.x or 16.x branch

File: tests/cl-go-client/go.mod
Comment (czarcas7ic) [inline tests/cl-go-client/go.mod:144]: same question

**Review by czarcas7ic — COMMENTED** [review]
It feels like we are missing some other gov protos, one that comes to mind is https://github.com/osmosis-labs/osmosis/blob/fd6707450c577e5fd1097608f730f350aa23e8a5/proto/osmosis/gamm/v1beta1/gov.proto#L14-L23

File: x/ibc-hooks/go.mod
Comment (czarcas7ic) [reply x/ibc-hooks/go.mod:146]: Sorry for the spam of "same question", wanted to ensure each location is marked so we can replace all of them with a commit in the correct branch if needed

File: go.mod
Comment (mattverse) [reply go.mod:318]: This is the commit https://github.com/osmosis-labs/cosmos-sdk/commit/14238df02ba5e9f75cc37a64df94bdef5fc5e42a

File: osmomath/go.mod
Comment (mattverse) [reply osmomath/go.mod:94]: same as above, fixing the comment to be 16.x though

File: tests/cl-go-client/go.mod
Comment (mattverse) [reply tests/cl-go-client/go.mod:144]: ditto for above

File: x/ibc-hooks/go.mod
Comment (mattverse) [reply x/ibc-hooks/go.mod:146]: no worries! And thank you for checking this! I think its always important that dependency changes are in the scope of review!

File: proto/amino/amino.proto
Comment (mattverse) [reply proto/amino/amino.proto:7]: must have been ported together from the cherry pick, erasing this

File: osmoutils/go.mod
Comment (mattverse) [reply osmoutils/go.mod:135]: same as above,

File: go.mod
Comment (mattverse) [reply go.mod:318]: Changing this to a more recent version of sdk v16.x, https://github.com/osmosis-labs/cosmos-sdk/commit/6ed904cdb75fe08a9310e7df49eaf9c5b2b64692

**Comment by pyramation** [comment]
excited for this! I've tested the cosmology fork and the new osmojs candidate release is working amazingly so far!

**Review by czarcas7ic — APPROVED** [review]
I don't see any risk in getting this merged. If you agree @mattverse feel free to hit merge on this

**Comment by czarcas7ic** [comment]
@mattverse is there any reason we cant merge this?

**Comment by mattverse** [comment]
@czarcas7ic No, there is not, has been stale for so long that i have forgotten this as well! Thanks for the review, merging this after fixing merge conflict

**Pull request merged**
